src/scrapers/poalim-business/helpers.js

```diff
@@ -0,0 +1,312 @@
+
+    const txnsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${accountNumber}&numItemsPerPage=200&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&sortCode=1`;
+    const txnsResult = await fetchPoalimXSRFWithinPage(page, txnsUrl, '/current-account/transactions');
+
+    let txns = [];
+    let checkingShekelsBalance = 0;
+    if (txnsResult) {
+      checkingShekelsBalance = txnsResult.transactions[0].currentBalance;
+
+      const date = new Date();
+      const dateString = new Date(date.getTime() - (date.getTimezoneOffset() * 60000))
+        .toISOString()
+        .split('T')[0];
+      fs.writeFile(`./old_data/POALIM_original_checking_bank_dump_${dateString}_${accountIndex}.json`, JSON.stringify(txnsResult.transactions), 'utf8', () => {
+        console.log('done dumping original checking dump file');
+      });
+
+      txns = convertTransactions(txnsResult.transactions);
+    }
+
+    let checkingDollarsBalance = 0;
+    let checkingDollarsBalanceInShekels = 0;
+    let checkingEuroBalance = 0;
+    let checkingEuroBalanceInShekels = 0;
+
+    const forigenTxnsUrl = `${apiSiteUrl}/foreign-currency/transactions?accountId=${accountNumber}&type=business&view=details&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&currencyCodeList=19,100&detailedAccountTypeCodeList=142&lang=he`;
+    const forigenTxnsResult = await fetchPoalimXSRFWithinPageGet(page, forigenTxnsUrl, '/foreign-currency/transactions');
+
+    const dollarsBalanceUrl = `${apiSiteUrl}/foreign-currency/transactions?accountId=${accountNumber}&view=graph&detailedAccountTypeCode=142&currencyCode=19&lang=he`;
+    const dollarsBalanceResult = await fetchPoalimXSRFWithinPageGet(page, dollarsBalanceUrl, '/foreign-currency/transactions');
+    const eurosBalanceUrl = `${apiSiteUrl}/foreign-currency/transactions?accountId=${accountNumber}&view=graph&detailedAccountTypeCode=142&currencyCode=100&lang=he`;
+    const eurosBalanceResult = await fetchPoalimXSRFWithinPageGet(page, eurosBalanceUrl, '/foreign-currency/transactions');
+
+    checkingDollarsBalance = dollarsBalanceResult.graphData[dollarsBalanceResult.graphData.length - 1].currencyAccountBalance;
+    checkingEuroBalance = eurosBalanceResult.graphData[eurosBalanceResult.graphData.length - 1].currencyAccountBalance;
+
+    const transactionBalanceUrl = `${apiSiteUrl}/foreign-currency/transactions?accountId=${accountNumber}&type=business&lang=he`;
+    const transactionBalanceResult = await fetchPoalimXSRFWithinPageGet(page, transactionBalanceUrl, '/foreign-currency/transactions');
+
+    for (
+      let foreignAccountbalancesIndex = 0;
+      foreignAccountbalancesIndex < transactionBalanceResult.balancesAndLimitsDataList.length;
+      foreignAccountbalancesIndex += 1) {
+      switch (transactionBalanceResult.balancesAndLimitsDataList[foreignAccountbalancesIndex].currencyCode) {
+        case 19:
```

Comment:
Those are for USD and Euro accounts.
I wonder if anyone has accounts with other currencies?